src/main/java/ch/njol/skript/expressions/base/EventValueExpression.java

```diff
@@ -77,20 +79,27 @@ public EventValueExpression(final Class<? extends T> c, final @Nullable Changer<
 		assert c != null;
 		this.c = c;
 		this.changer = changer;
+		single = !c.isArray();
+		componentType = single ? c : c.getComponentType();
 	}
 	
 	@Override
 	@Nullable
+	@SuppressWarnings({"unchecked", "SuspiciousSystemArraycopy"})
 	protected T[] get(final Event e) {
 		final T o = getValue(e);
 		if (o == null)
-			return null;
-		@SuppressWarnings("unchecked")
-		final T[] one = (T[]) Array.newInstance(c, 1);
-		one[0] = o;
-		return one;
+			return (T[]) new Object[0];
+		if (single) {
+			final T[] one = (T[]) Array.newInstance(c, 1);
+			one[0] = o;
+			return one;
+		}
+		T[] array = (T[]) Array.newInstance(c.getComponentType(), ((T[]) o).length);
+		System.arraycopy(o, 0, array, 0, array.length);
```

Comment:
```suggestion
		T[] dataArray = (T[]) o;
		T[] array = (T[]) Array.newInstance(c.getComponentType(), dataArray.length);
		System.arraycopy(dataArray, 0, array, 0, array.length);
```
This will avoid the SuspiciousSystemArraycopy warning.